### DISCUSSION TITLE
Would it be possible moving from OBJ-C -> Swift when dealing with metal in MoltenVK?

### DISCUSSION THREAD

**Discussion by italomandara** [KhronosGroup/MoltenVK#1662] [Ideas]
The point is that Swift is claimed to be a faster language and also easier to use so more devs could help in the project?

**Comment by cdavis5e** [comment]
Swift uses ARC, and unlike in Objective-C, it cannot be turned off. My understanding is that MoltenVK deliberately uses MRR because the retain and release calls inserted by ARC, some of which are extraneous and unnecessary, slow it down too much.

Also, since Swift does not yet have interop with C++ (only with C and Objective-C), rewriting MoltenVK in Swift would be a humongous undertaking, one that I'm not sure is worth the cost.

**Comment by billhollings** [comment]
Yeah. What he (@cdavis5e) said.

Interestingly, Apple has released [`metal-cpp`](https://developer.apple.com/videos/play/wwdc2022/10160/), which is a C++ binding for Metal, and uses MRC for the underlying Objective-C Metal objects. If we did want to make the Metal code more accessible to contributors who are already familiar with C++, that would probably be the way to do it.

**Reply by italomandara** [reply]
I think most developers would rather learn C++ than Objective-C (or already know C++) , also C++ is faster

**Comment by billhollings** [comment]
> also C++ is faster

Well...Metal is still written in Objective-C. Both the Swift bindings, and Apple's new C++ bindings, are wrappers around that.

But nevertheless...evolving to use the C++ wrapper within MoltenVK might make the MoltenVK code base more accessible to contributors coming from elsewhere.
